Implementing color change in the selected item from ListInput

fgemig opened the pull request:
## Summary
The proposed change makes it possible to change the color in the selected ListInput item. 

## Motivation
This change improves the way the user views the selected items in the list, avoiding choosing an incorrect item.

## Example
![exemplo-inquirer](https://user-images.githubusercontent.com/23200856/95026720-b866a700-0669-11eb-8d57-0e4065a59545.gif)

Well, I hope that this implementation will be useful for the community. I am open to questions and accept feedbacks!

Commit:
Implementing color change in the selected item from ListInput

Commit:
Implementing color change in the selected item from ListInput

afucher commented on src/Inquirer/Prompts/ListInputMessage.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace InquirerCore.Prompts
+{
+    public class ListInputMessage
```
Instead of creating a ListInputMessage, it can be a ConsoleMessage or just a Message to create a more generic class that will allow us to use in more places.
It can just have now the string message and the color, and the logic related to the color can be outside the class.

fgemig replied:
Obrigado pelo feedback! Você tem razão. Uma classe mais genérica permitiria o uso em mais lugares e não exclusivamente no ListInput. Estarei revendo essa mudança no próximo PR.

fgemig replied:
Thanks for the feedback! You're right. A more generic class would allow use in more places and not exclusively on ListInput. I will be reviewing this change in the next PR.

afucher commented on src/Inquirer/Prompts/IScreenManager.cs:
```diff
@@ -7,6 +8,7 @@ public interface IScreenManager
     {
         int[,] RenderMultipleMessages(string[] messages);
         int[] Render(string[] content, string[] bottomContent);
+        int[] RenderList(ListInputMessage[] content, string[] bottomContent);
```
I prefer to not have a specific render to list, so it can be just a Render.

fgemig replied:
Great! I will try to implement only with the Render method instead of having an exclusive for ListInput.

afucher commented on test/InquirerUnitTest/InputListUnitTest.cs:
```diff
@@ -62,8 +62,41 @@ public void ShouldCallConsoleWriteLine()
+        }
+
+        [Fact]
+        public void FirstQuestionNotShoulBeADifferentColor()
```
I think here is related to the Second question and not First, right?

fgemig replied:
Yes, you are correct! The method name refers to the second item in the list.

afucher reviewed (changes requested):
Hey @fgemig thanks for your PR!

I've added some feedbacks to make the solution more generic, and allow us to use in other places in the future.
Please let me know what do you think.

Joaolfelicio committed:
Added CreditCardNumberValidator and Unit Testing for it. #10 (#19)

* Added CreditCardNumberValidator and Unit Testing for it. #10

* Updated class name from CreditCardValidator to CreditCardNumberValidator

* Implemented Luhn Algorithm, added more Unit Tests

* Updated regex to only contain digits and to have a length of between 8 and 19 digits.

Credit Card Numbers are composed of 8 to 19 digits (Source: https://en.wikipedia.org/wiki/Payment_card_number).

* Updated readme.

Updated error message for credit card validation.

* Split unit testing methods by type of refusal.

* Inverted valid number format logic

Co-authored-by: Arthur Fücher <arthurfucher@gmail.com>

Co-authored-by: Arthur Fücher <arthurfucher@gmail.com>

Reisclef committed:
#10 Add PercentageValidator and tests (#14)

* Add PercentageValidator and tests

* Allowing percentage symbol, changed error message

* Added additional unit tests

* Only allow one percentage symbol at end of input

Commit:
Merge branch 'dev' of https://github.com/fgemig/Inquirer into dev

Commit:
Refactoring the ListInput method to change the color of the selected item
